ldes-fragmentisers/ldes-fragmentisers-geospatial/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/geospatial/GeospatialFragmentationService.java

```diff
@@ -29,45 +31,58 @@ public class GeospatialFragmentationService extends FragmentationServiceDecorato
 	private final LdesFragmentRepository ldesFragmentRepository;
 	private final FragmentCreator fragmentCreator;
 	private final GeospatialBucketiser geospatialBucketiser;
+	private final Tracer tracer;
+	private Span rootSpan;
 
 	public GeospatialFragmentationService(FragmentationService fragmentationService, LdesConfig ldesConfig,
 			LdesMemberRepository ldesMemberRepository,
 			LdesFragmentRepository ldesFragmentRepository, FragmentCreator fragmentCreator,
-			GeospatialBucketiser geospatialBucketiser) {
+			GeospatialBucketiser geospatialBucketiser, Tracer tracer) {
 		super(fragmentationService, ldesFragmentRepository);
 		this.ldesConfig = ldesConfig;
 		this.ldesMemberRepository = ldesMemberRepository;
 		this.ldesFragmentRepository = ldesFragmentRepository;
 		this.fragmentCreator = fragmentCreator;
 		this.geospatialBucketiser = geospatialBucketiser;
+		this.tracer = tracer;
 	}
 
 	@Override
 	public void addMemberToFragment(LdesFragment parentFragment, String ldesMemberId) {
+		rootSpan = this.tracer.nextSpan().name("Geobased fragmentation").start();
+		Span bucketisingSpan = this.tracer.nextSpan(rootSpan).name("Member Bucketising").start();
```

Comment:
would limit to 1 span per method.

Reply:
In this case. I'd add a span in the bucketise method instead of defining the bucketisingSpan here.

Reply:
Would be an option.. but it might cause problems with getting the parent Span as I found previously